### PR TITLE
KEYCLOAK-10744 Fix MariaDB cannot create database

### PR DESCRIPTION
**Pull request by hmlnarik**

### CODE REVIEW

**Comment by hmlnarik** [comment]
Can be tested via
`mvn install -f testsuite/integration-arquillian/pom.xml -Dtest=UserTest\#verifyCreateUser -Pdb-mariadb -Ddocker.database.image=mariadb:10.3.16 -Dmariadb.version=2.4.2 -fae`

File: model/jpa/src/main/java/org/keycloak/connections/jpa/updater/liquibase/UpdatedMariaDBDatabase.java [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+import java.util.HashSet;
+import java.util.Set;
+import liquibase.database.core.MariaDBDatabase;
+
+/**
+ * @author <a href="mailto:mposolda@redhat.com">Marek Posolda</a>
+ */
+public class UpdatedMariaDBDatabase extends MariaDBDatabase {
+
+    private static final Set<String> RESERVED_WORDS = new HashSet();
```
Comment (vramik) [inline model/jpa/src/main/java/org/keycloak/connections/jpa/updater/liquibase/UpdatedMariaDBDatabase.java]: unchecked conversion:
```suggestion
    private static final Set<String> RESERVED_WORDS = new HashSet<>();
```

File: model/jpa/src/main/java/org/keycloak/connections/jpa/updater/liquibase/UpdatedMariaDBDatabase.java [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+    @Override
+    public boolean isReservedWord(String string) {
+        return super.isReservedWord(string) || RESERVED_WORDS.contains(string.toUpperCase());
+    }
+
+    @Override
+    public int getPriority() {
+        return super.getPriority() + 1; // Always take precedence over factory MySQLDatabase
```
Comment (vramik) [inline model/jpa/src/main/java/org/keycloak/connections/jpa/updater/liquibase/UpdatedMariaDBDatabase.java]:
```suggestion
        return super.getPriority() + 1; // Always take precedence over factory MariaDBDatabase
```

File: model/jpa/src/main/java/org/keycloak/connections/jpa/updater/liquibase/UpdatedMariaDBDatabase.java [hunk 3]
```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright 2016 Red Hat, Inc. and/or its affiliates
```
Comment (vramik) [inline model/jpa/src/main/java/org/keycloak/connections/jpa/updater/liquibase/UpdatedMariaDBDatabase.java]:
```suggestion
 * Copyright 2019 Red Hat, Inc. and/or its affiliates
```

**Review by vramik — DISMISSED** [review]
@hmlnarik thank you for the PR, I've found just few nits, feel free to ignore them. It's good to go in as it is.

**Comment by hmlnarik** [comment]
Thanks @vramik ! Fixed.

**Review by vramik — APPROVED** [review]

**Review by abstractj — APPROVED** [review]

**Pull request merged**